iroh-sync/src/store/memory.rs

```diff
@@ -509,43 +454,51 @@ impl crate::ranger::Store<RecordIdentifier, SignedEntry> for ReplicaStoreInstanc
     }
 
     fn put(&mut self, k: RecordIdentifier, v: SignedEntry) -> Result<(), Self::Error> {
-        // TODO: propagate error/not insertion?
+        assert_eq!(k.timestamp(), v.timestamp(), "inconsistent timestamp"); // TODO: error
+                                                                            // TODO: propagate error/not insertion?
         if v.verify().is_ok() {
-            let timestamp = v.entry().record().timestamp();
             // TODO: verify timestamp is "reasonable"
```

Comment:
given the amount of reliance we have now on this, we should really do some sanity checks here

Reply:
Hm, what kinds of sanity checks are you thinking about? Timestamp not being too far in the future from the device's wall clock?

Reply:
> Timestamp not being too far in the future from the device's wall clock?

yes: for reference: https://willowprotocol.org/more/timestamps-really

the suggestion is 10min into the future